types/src/v0/impls/auction.rs

```diff
@@ -0,0 +1,274 @@
+        let comm = committable::RawCommitmentBuilder::new(&Self::tag())
+            .fixed_size_field("account", &self.account.to_fixed_bytes())
+            .fixed_size_field("gas_price", &self.gas_price.to_fixed_bytes())
+            .fixed_size_field("bid_amount", &self.bid_amount.to_fixed_bytes())
+            .var_size_field("url", self.url.as_str().as_ref())
+            .u64_field("view", self.view.u64())
+            .var_size_field("namespaces", &bincode::serialize(&self.namespaces).unwrap());
+        comm.finalize()
+    }
+}
+
+impl BidTxBody {
+    /// Construct a new `BidTxBody`.
+    pub fn new(
+        account: FeeAccount,
+        bid: FeeAmount,
+        view: ViewNumber,
+        namespaces: Vec<NamespaceId>,
+        url: Url,
+    ) -> Self {
+        Self {
+            account,
+            bid_amount: bid,
+            view,
+            namespaces,
+            url,
+            // TODO gas_price will come from config probably, but we
+            // can use any value for first round of integration
+            ..Self::default()
+        }
+    }
+
+    /// Sign `BidTxBody` and return the signature.
+    pub fn sign(&self, key: &EthKeyPair) -> Result<Signature, SigningError> {
+        FeeAccount::sign_builder_message(key, self.commit().as_ref())
+    }
+    /// Sign Body and return a `BidTx`. This is the expected way to obtain a `BidTx`.
+    /// ```
+    /// # use espresso_types::FeeAccount;
+    /// # use espresso_types::v0_3::BidTxBody;
+    ///
+    /// let key = FeeAccount::test_key_pair();
+    /// BidTxBody::default().signed(&key).unwrap();
+    /// ```
+    pub fn signed(self, key: &EthKeyPair) -> Result<BidTx, SigningError> {
+        let signature = self.sign(key)?;
+        let bid = BidTx {
+            body: self,
+            signature,
+        };
+        Ok(bid)
+    }
+
+    /// Get account submitting the bid
+    pub fn account(&self) -> FeeAccount {
+        self.account
+    }
+    /// Get amount of bid
+    pub fn amount(&self) -> FeeAmount {
+        self.bid_amount
+    }
+    /// Update `url` field on a previously instantiated `BidTxBody`.
+    pub fn with_url(self, url: Url) -> Self {
+        Self { url, ..self }
+    }
+}
+
+impl Default for BidTxBody {
+    fn default() -> Self {
+        let key = FeeAccount::test_key_pair();
+        let nsid = NamespaceId::from(999u64);
+        Self {
+            url: Url::from_str("https://sequencer:3939").unwrap(),
+            account: key.fee_account(),
+            gas_price: FeeAmount::default(),
+            bid_amount: FeeAmount::default(),
+            view: ViewNumber::genesis(),
+            namespaces: vec![nsid],
+        }
+    }
+}
+impl Default for BidTx {
+    fn default() -> Self {
+        let body = BidTxBody::default();
+        let key = FeeAccount::test_key_pair();
+        let signature = FeeAccount::sign_builder_message(&key, body.commit().as_ref()).unwrap();
+        Self { signature, body }
```

Comment:
Nit: Can we replace these lines with the `signed` function?

Reply:
e0fdf330b8d1133bebf540538bb40737a77c1713